sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Metrics/ExecutableLinesWalkerTest.cs

```diff
@@ -328,5 +328,24 @@ static void Main(string[] args)
                 }",
                 4);
         }
+
+        [TestMethod]
+        public void Test_18_ExcludeFromTestCoverage()
+        {
+            AssertLinesOfCode(
+              @"using System.Diagnostics.CodeAnalysis;
+                namespace project_1
+                {
+                    public class ComplicatedCode
+                    {
+                        [ExcludeFromCodeCoverage]

```

Comment:
It would be worth testing some of the other valid forms of attribute naming e.g. 
    [ExcludeFromCodeCoverageAttribute]
    [ExcludeFromCodeCoverage()]
    [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage()]

Reply:
https://msdn.microsoft.com/library/system.diagnostics.codeanalysis.excludefromcodecoverageattribute.aspx
mentions:

> Placing this attribute on a class or a structure excludes all the members of that class or structure from the collection of code coverage information.

This appears to be a per-class attribute, not a per-method attribute (or not only a per-method attribute). I think the current code does not take that case into account.

Reply:
Hi @MaxTranced,

As you can see with the `AttributeUsageAttribute` it can be set on class, struct, ctor, method, property and events. You are right when you say that when put on class or struct it excludes everything but we do have tests for that (see `Test_20_ExcludeClassFromTestCoverage` and `Test_21_ExcludeStructFromTestCoverage`).

Reply:
Hi @Evangelink 

Thank you for the answer... Then I guess I've probably hit a bug. The code I have (that is not excluded from the code coverage analysis) is roughly:

```
    [ExcludeFromCodeCoverage]
    [CLSCompliant(true)]
    public static class LoggerFactory
    {
```

I will file an issue.